### ISSUE TITLE
Apply validation if checkbox is checked

### ISSUE THREAD

**Issue by andriezzo** [horprogs/Just-validate#30]
I would like to validate some fields (with separate rules, messages etc) as usual... but only after a checkbox is checked. 
I tried with some if...else but I think I missed something. 
(Hope this could be a feature in the next releases)

Any suggestions? Thanks in advance

**Comment by horprogs** [comment]
Hi! Do you mean you want to show validation errors right after fields changing? Without clicking on the submit button?

**Comment by andriezzo** [comment]
No, I want to show validation errors after submitting the form. 
But it would be cool to have them also in real-time.

Just to be clear, I have two text inputs and a checkbox
input1   (not required)
input2 (not required)
checkbox (not required) 

if the checkbox is checked, I want the input to become required immediately, without submitting the form.
So when I click the button, the errors become visible. 

Thank you! :)

**Comment by horprogs** [comment]
Hmm, now the errors showing in real-time, after you submit the form. For example, you have:
input1 (required)
input2 (required)
checkbox (required)

Initially we don't show the errors. You submit the form, the errors for all invalid fields are visible. If you click on a checkbox, the error will be hidden (for example here https://codepen.io/Horprogs/pen/abLyVmP)

In your case if all fields are not required you wouldn't get the errors at all.

I guess it would be better to create a reproducible example (it's easier to do it based on this pen https://codepen.io/Horprogs/pen/abLyVmP) and I will try to help

**Comment by andriezzo** [comment]
My idea is to have a checkbox that applies the "required" attribute on some input elements. 
Will try to do something on your Codepen but I'm afraid for not having enough experience with JS :(

**Comment by horprogs** [comment]
It seems I got you, let me try something

**Comment by horprogs** [comment]
Just to clarify, you want to be able to change/add new rules by condition, like this: https://codepen.io/Horprogs/pen/BawdYoE

I will add `.removeField()` method, so you will be able to remove and add fields based on some logic

**Comment by andriezzo** [comment]
> Just to clarify, you want to be able to change/add new rules by condition, like this: https://codepen.io/Horprogs/pen/BawdYoE

That's exactly what I wanted! :) 
And it works perfectly even if the checkbox is not required. 

> I will add `.removeField()` method, so you will be able to remove and add fields based on some logic

I'm so happy you helped me and I hope this feature will be useful to other users!
Thank you!

**Comment by horprogs** [comment]
Thanks for the suggestion!
Released 2.1.0 version, you could check the example here: https://just-validate.dev/examples/#conditional

Initially, the name field is not required, then you click on the checkbox and the name field becomes required. The code example:

```
const validation = new JustValidate("#conditional-validation-form", {
  errorFieldCssClass: "is-invalid",
  errorLabelStyle: {
    fontSize: "14px",
    color: "#dc3545",
  },
  focusInvalidField: true,
  lockForm: true,
});

validation.onSuccess((ev) => {
  ev.preventDefault();
  window.showNotification();
});

document.querySelector("#example6_checkbox").addEventListener("click", (e) => {
  if (e.target.checked) {
    validation.addField("#example6_name", [
      {
        rule: "required",
      },
    ]);
  } else {
    validation.removeField("#example6_name");
  }
});
```

**Comment by andriezzo** [comment]
Wow fantastic, I'm glad my request was useful :) 
I recently managed to set up a confirmation modal that appears on form submit only if all fields are valid. 
Maybe it's not related but...
